src/clients/FlexbaseClient.Company.ts

```diff
@@ -30,12 +35,18 @@ export class FlexbaseClientCompany extends FlexbaseClientBase {
         }
     }
 
-    async getCompanyPayments(): Promise<Payment[]> {
+    async getCompanyPayments(): Promise<Payment[] | null> {
```

Comment:
I think putting a `null` response isn’t a good idea, we need good feedback if the response fails, and logged it, and I think we need to return the `success` value, where on the web repository we can manage these states.
You should leave the `promise` with the `PaymentsResponse`

```
    async getCompanyPayments(): Promise<PaymentsResponse> {
        try {
            const response = await this.client.url('/servicing/payments').get().json<PaymentsResponse>();
            if (!response.success) {
              this.logger.error('Some message error here', response.error);
          }
            return response;
        } catch (error) {
            this.logger.error(`Unable to get company payments`, error);
            return { success: false, error };
        }
    }
```